create-mobile-geodatabase/src/main/java/com/esri/arcgismaps/sample/createmobilegeodatabase/MainActivity.kt

```diff
@@ -0,0 +1,300 @@
+            startActivity(geodatabaseIntent)
+        }
+    }
+
+
+    /**
+     * Create and load a new geodatabase file with TableDescription fields
+     */
+    private fun createGeodatabase() {
+        // define the path and name of the geodatabase file
+        // note: the path defined must be non-empty, available,
+        // allow read/write access, and end in ".geodatabase"
+        val file = File(getExternalFilesDir(null)?.path + "/LocationHistory.geodatabase")
+        if (file.exists()) {
+            file.delete()
+        }
+        // close the existing geodatabase
+        geodatabase?.close()
+        lifecycleScope.launch {
+            // create a geodatabase file at the file path
+            Geodatabase.create(file.path).onSuccess { geodatabase ->
+                // keep the instance of the new geodatabase for sharing
+                this@MainActivity.geodatabase = geodatabase
+                createGeodatabaseFeatureTable()
+            }.onFailure {
+                showError(it.message.toString())
+            }
+
+        }
+    }
+
+    /**
+     * Create a new [featureTable] using a custom table description
+     * and add the feature layer to the MapView.
+     */
+    private suspend fun createGeodatabaseFeatureTable() {
+        // construct a table description which stores features as points on map
+        val tableDescription =
+            TableDescription(
+                "LocationHistory",
+                SpatialReference.wgs84(),
+                GeometryType.Point
+            )
+        // set up the fields to the table,
+        // Field.Type.OID is the primary key of the SQLite table
+        // Field.Type.DATE is a date column used to store a Calendar date
+        // FieldDescriptions can be a SHORT, INTEGER, GUID, FLOAT, DOUBLE, DATE, TEXT, OID, GLOBALID, BLOB, GEOMETRY, RASTER, or XML.
+        tableDescription.fieldDescriptions.addAll(
+            listOf(
+                FieldDescription("oid", FieldType.Oid),
+                FieldDescription("collection_timestamp", FieldType.Date)
+            )
+        )
+
+        // set any properties not needed to false
+        tableDescription.apply {
+            hasAttachments = false
+            hasM = false
+            hasZ = false
+        }
+
+        // add a new table to the geodatabase by creating one from the tableDescription
+        geodatabase?.createTable(tableDescription)?.onSuccess { featureTable ->
+            // get the result of the loaded "LocationHistory" table
+            this.featureTable = featureTable
+            // create a feature layer for the map using the GeodatabaseFeatureTable
+            val featureLayer = FeatureLayer(featureTable)
+            mapView.map?.operationalLayers?.add(featureLayer)
+            // display the current count of features in the FeatureTable
+            featureCountTextView.text =
+                "Number of features added: ${featureTable?.numberOfFeatures}"
```

Comment:
My IDE tells me that you don't need the `?`
```suggestion
                "Number of features added: ${featureTable.numberOfFeatures}"
```